### PR TITLE
WIP for prototype of L1 database

### PR DESCRIPTION
**Pull request by andy-slac**
Start implementation of the L1 database prototype. Few simple things are implemented:
- Python script to load L1db schema into mysql (this may also run with postgresql, not tested)
- generators for sources to be used later in simuation

### CODE REVIEW

File: bin/create_l1_schema [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    # configure logging
+    _configLogger(args.verbose)
+
+    # instantiate db engine
+    engine = engineFactory.getEngineFromFile(args.config)
+    logging.info('engine name={} driver={}'.format(engine.name, engine.driver))
+
+    # metadata (DDL) instance
+    metadata = MetaData(engine)
+    mysql_engine = 'InnoDB'
```
Comment (vaikuntht) [inline bin/create_l1_schema]: Aren't we using MyISAM? Or is InnoDB only for L1db?

File: bin/create_l1_schema [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    DOUBLE = make_doube_type(engine)
+    FLOAT = sqlalchemy.types.Float
+    DATETIME = sqlalchemy.types.TIMESTAMP
+    BIGINT = sqlalchemy.types.BigInteger
+    INTEGER = INT = sqlalchemy.types.Integer
+    BLOB = sqlalchemy.types.BLOB
+    CHAR = sqlalchemy.types.CHAR
+
+    diaObject = Table('DiaObject', metadata,
+        Column('diaObjectId', BIGINT, nullable=False),
```
Comment (vaikuntht) [inline bin/create_l1_schema]: Aren't we using ObjecID as primary keys?

Comment (vaikuntht) [inline bin/create_l1_schema]: Ok nvm, I see it at the bottom of the schemas.

File: bin/gen_sources [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#!/bin/env python
+
+""" Application to generate a list of random positions for sources.
+
+Generates and saves the list of sky coordiantes for the sources to
```
Comment (vaikuntht) [inline bin/gen_sources]: co-ordiantes -> co-ordinates

File: bin/gen_sources [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+#---------------------------------
+
+def main():
+
+    descr = 'One-line application description.'
+    parser = ArgumentParser(description=descr)
+    parser.add_argument('-v', '--verbose', action='count', default=0,
+                        help='More verbose output, can use several times.')
+    parser.add_argument('-m', '--mode', action='store', default='xyz',
+                        help='Defines type of output data, possible valus are "xyz", def: "xyz"')
```
Comment (vaikuntht) [inline bin/gen_sources]: valus -> values

File: bin/gen_sources [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    parser.add_argument('-H', '--hemi', type=int, default=0,
+                        help='Negative for southern hemisphere, positive for nothern, def: 0')
```
Comment (vaikuntht) [inline bin/gen_sources]: nothern -> northern

Comment (vaikuntht) [inline bin/gen_sources]: Shouldn't this be a required argument then since 0 is neither positive or negative?

File: bin/create_l1_schema (hunk 1 above)
Comment (andy-slac) [inline bin/create_l1_schema]: It is for L1 only, we do need transactions there.

File: bin/gen_sources (hunk 3 above)
Comment (andy-slac) [inline bin/gen_sources]: OK, even though `coordiantes` has plenty of hits in google :smile:

File: bin/gen_sources (hunk 5 above)
Comment (andy-slac) [inline bin/gen_sources]: Default 0 covers whole sphere, I'll extend help message.

File: bin/gen_sources [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                        help='Number of sources per visit, def: 10000')
+    parser.add_argument('file', help='Name of output file')
+    args = parser.parse_args()
+
+    # configure logging
+    _configLogger(args.verbose)
+
+    # total number of sources
+    area = 2 * math.pi if args.hemi != 0 else 4 * math.pi
+    visit_area = (FOV / 2 * math.pi / 180) ** 2 * math.pi
```
Comment (vaikuntht) [inline bin/gen_sources]: This confused me as I thought it was (number)^(2pi) . Easier for readability if you use parenthesis to indicate order of operations.

Comment (andy-slac) [inline bin/gen_sources]: OK, I'll change to:

    visit_area = math.pi * (FOV / 2 * math.pi / 180) ** 2

File: python/lsst/l1dbproto/generators.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    """
+
+    v = numpy.cross(a, b)
+    sin = numpy.linalg.norm(v)
+    if sin == 0:
+        return numpy.identity(3)
+    cos = numpy.vdot(a, b)
+    vx = numpy.mat([[0, -v[2], v[1]], [v[2], 0., -v[0]], [-v[1], v[0], 0.]])
+
+    R = numpy.identity(3) + vx + vx * vx * (1 - cos) / sin ** 2
```
Comment (vaikuntht) [inline python/lsst/l1dbproto/generators.py]: Same comment here, grouping of terms for readability.

Comment (andy-slac) [inline python/lsst/l1dbproto/generators.py]: OK

File: tests/test_generators.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
+#!/bin/env python
+
+"""
+Brief one-line description of the script.
+
+Following paragraphs provide detailed description of the module, its
+contents and usage.
```
Comment (vaikuntht) [inline tests/test_generators.py]: Template leftover?

Comment (andy-slac) [inline tests/test_generators.py]: Indeed, will replace with something reasonable.

File: bin/gen_sources (hunk 3 above)
Comment (vaikuntht) [inline bin/gen_sources]: Haha.

**Pull request merged**